### ISSUE TITLE
GetVirtualMachineUserData returns nil / Correct usage?

### ISSUE THREAD

**Issue by jescarri** [xanzy/go-cloudstack#122]
Hi I've been trying to make a simple API Call to get the instance user-data however I get a "nil" response.

Client is set as follows:

````
        c := cs.NewAsyncClient(apiurl, apikey, secret, false)
        c.HTTPGETOnly = false
        c.AsyncTimeout(6000)
````

Call to get the User-data

````
        getUserDataParams := c.Client.User.NewGetVirtualMachineUserDataParams("a71c7403-a242-4f29-88fd-c6e49fc9249b")
        response, err := c.User.GetVirtualMachineUserData(getUserDataParams)
        spew.Dump(getUserDataParams)
        spew.Dump(response)

````

Response on GoCode:

````
(*cloudstack.GetVirtualMachineUserDataParams)(0xc00000e288)({
 p: (map[string]interface {}) (len=1) {
  (string) (len=16) "virtualmachineid": (string) (len=36) "a71c7403-a242-4f29-88fd-c6e49fc9249b"
 }
})
(*cloudstack.GetVirtualMachineUserDataResponse)(0xc0001a0140)({
 JobID: (string) "",
 Jobstatus: (int) 0,
 Userdata: (string) "",
 Virtualmachineid: (string) ""
})
````

Logs on api-server

````
2019-10-15 00:23:04,515 INFO  [a.c.c.a.ApiServer] (qtp895947612-122991:ctx-78d594d8 ctx-dea73ea6 ctx-edba6901) (logid:d35e971d) (userId=2 accountId=2 sessionId=null) 127.0.0.1 -- GET apiKey=EDITED&command=getVirtualMachineUserData&response=json&virtualmachineid=a71c7403-a242-4f29-88fd-c6e49fc9249b&signature=EDITED 200 {"getvirtualmachineuserdataresponse":{"virtualmachineuserdata":{"virtualmachineid":"a71c7403-a242-4f29-88fd-c6e49fc9249b","userdata":"EDITED"}}}
````

If I do the same api-call on cloudmonkey I get the same 200 response code from api but cloudmonkey does get the user-data.

I'm doing anything wrong?
Client version is v2.7.0
ApiVersion is: 4.11.2.0

Thanks!

**Comment by svanharmelen** [comment]
Does the error contain any info? It seems you are getting an async job response instead of the actual response which should be an error in my opinion. Not sure if this is related to the CS API or this package to be honest.

**Comment by jescarri** [comment]
@svanharmelen sadly it does not return any error.

**Comment by jescarri** [comment]
as per the code, the cloudstack.GetVirtualMachineUserDataResponse{} type has those async-job fields, is kinda weird, anything else should I be checking?

**Comment by svanharmelen** [comment]
Could you try the code in this branch: https://github.com/xanzy/go-cloudstack/tree/svh/b-user-data

I think this should solve your issue.

**Comment by jescarri** [comment]
Sure! thanks I will test it in a bit and report back!

Thanks!

**Comment by jescarri** [comment]
@svanharmelen I can confirm your fix works! thanks!

**Comment by svanharmelen** [comment]
Fixed by PR #123 and released in version v2.7.1
